cs_api_cli/cs_api_cli.ml

```diff
@@ -16,10 +16,30 @@ let get_file path =
       | _ ->
         Lwt_result.fail ("Could not read file " ^ path ^ " for unknown reasons"))
 
+let resolve_project_name ~api ~verify ~project_id ~project_name =
+  (* The user can provide an ID or a name. If a name is provided, look for the
+     corresponding ID. Otherwise, just return the given ID. *)
+  let open Lwt_result.Infix in
+  match (project_id, project_name) with
+  | (None, None)
+  | (Some _, Some _) ->
+    Lwt_result.fail "Exactly one of project ID or name must be provided."
+  | (Some id, None) -> Lwt_result.return id
+  | (None, Some name) -> (
+    Cs_api_core.build_list_projects_request ~api
+    |> Cs_api_io.send_request ~verify
+    >>= Cs_api_io.get_response
+    >>= fun body ->
+    let projects = Cs_api_core.parse_list_projects_response ~body in
+    match CCList.Assoc.get ~eq:String.equal name projects with
+    | None -> Lwt_result.fail "Project not found"
```

Comment:
It's a tiny point, but perhaps it would be more helpful to say `"Project \"" ^ name ^ "\" not found"` or something like that?